homeassistant/components/lutron/config_flow.py

```diff
@@ -0,0 +1,89 @@
+"""Config flow to configure the Lutron integration."""
+from __future__ import annotations
+
+import logging
+from urllib.error import HTTPError
+
+from pylutron import Lutron
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LutronConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """User prompt for Main Repeater configuration information."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """First step in the config flow."""
+        errors = {}
+
+        if user_input is not None:
+            # Validate user input
+            ip_address = user_input[CONF_HOST]
+
+            # Now that we have the data, check and see if we can connect and get a GUID
+            main_repeater = Lutron(
+                ip_address,
+                user_input.get(CONF_USERNAME),
+                user_input.get(CONF_PASSWORD),
+            )
+
+            def _load_db() -> bool:
+                main_repeater.load_xml_db()
+                return True
+
+            try:
+                await self.hass.async_add_executor_job(_load_db)
+                guid = main_repeater.guid
```

Comment:
I think this is not raising anything, so this can be moved outside of the try block

Reply:
this can throw an HTTPerror which I've imported.  the unknown exception was some initial code before I found the actual exception being thrown so I can remove that class.

Reply:
I mean the .guid part

Reply:
already moved to the right part